django/forms/models.py

```diff
@@ -1184,8 +1184,11 @@ class ModelChoiceField(ChoiceField):
     def __init__(self, queryset, *, empty_label="---------",
                  required=True, widget=None, label=None, initial=None,
                  help_text='', to_field_name=None, limit_choices_to=None,
-                 **kwargs):
-        if required and (initial is not None):
+                 blank=False, **kwargs):
+        if (
+            required and initial is not None or
+            (isinstance(widget, RadioSelect) and blank is True and empty_label != _('None'))
```

Comment:
IMO this doesn't work properly, because it removes an empty option from `RadioSelect()` if `blank` **is allowed** which is incorrect. I also don't understand why we're checking `empty_label` :thinking:. I would expect sth similar to:
```python
(required and (initial is not None)) or
(isinstance(widget, RadioSelect) and not blank)
```

Reply:
Yes, you are right. It was my fault, I understood and implemented it completely in the opposite way.